### ISSUE TITLE
I am using qbittorrent to test my code. I am not sure if it is my code problem or my QB is not configured correctly

### ISSUE THREAD

**Issue by jimmyduang** [anacrolix/torrent#753]
When I open **QB** to start downloading files, the download speed is 0, because there is no seed service. 
Then when I keep starting the seed service, **QB** can instantly download some chunks, but then the download speed becomes 0 bytes.
this is video:

https://user-images.githubusercontent.com/34672876/172748867-a7f32707-e076-4ab7-998d-252045cfb791.mp4

and . this is my seed service code:

```package main

import (
	"fmt"
	"github.com/anacrolix/log"
	"github.com/anacrolix/torrent"
	"github.com/anacrolix/torrent/bencode"
	"github.com/anacrolix/torrent/metainfo"
	"net"
	"net/url"
	"os"
	"path/filepath"
	"strings"
)

type TorrentConfig struct {
	PieceSize    int64
	AnnounceList metainfo.AnnounceList
}

func Make(filePath string, config *TorrentConfig, torrentsDir string) (*metainfo.MetaInfo, error) {
	// 获取种子配置;
	if config == nil {
		config = &TorrentConfig{
			PieceSize:    512 * 1024,
			AnnounceList: [][]string{{"udp://192.168.2.181:6969"}},
		}
	}

	metaInfo := &metainfo.MetaInfo{
		AnnounceList: config.AnnounceList,
	}
	if (len(metaInfo.AnnounceList) == 1) && (len(metaInfo.AnnounceList[0]) == 1) {
		metaInfo.Announce = metaInfo.AnnounceList[0][0]
		metaInfo.AnnounceList = nil
	}
	metaInfo.SetDefaults()
	info := metainfo.Info{
		PieceLength: config.PieceSize,
	}
	// 开始做种;
	err := info.BuildFromFilePath(filePath)
	if err != nil {
		return nil, err
	}
	metaInfo.InfoBytes, err = bencode.Marshal(info)
	if err != nil {
		return nil, err
	}

	// 存放种子文件;
	_ = os.MkdirAll(torrentsDir, os.ModeDir|os.ModePerm)
	infoHash := metaInfo.HashInfoBytes().HexString()
	infoHash = strings.ToLower(infoHash)
	torrentPath := filepath.Join(torrentsDir, infoHash+".torrent")
	log.Printf("创建新文件种子文件: %s - %s", filePath, torrentPath)
	f, err := os.Create(torrentPath)
	if err != nil {
		return nil, err
	}
	defer f.Close()

	err = metaInfo.Write(f)
	if err != nil {
		return nil, err
	}

	return metaInfo, nil
}
func AddFile(infoHash *metainfo.MetaInfo, filePath string) error {
	cfg := torrent.NewDefaultClientConfig()
	cfg.Seed = true
	cfg.NoDHT = true
	cfg.Debug = true
	cfg.Logger = log.Default
	cfg.DataDir = filePath

	//uploadLimiter := rate.NewLimiter(128*10*bytes.MB, 16*bytes.KB)
	//cfg.UploadRateLimiter = uploadLimiter

	//downloadLimiter := rate.NewLimiter(128*10*bytes.MB, 16*bytes.KB)
	//cfg.DownloadRateLimiter = downloadLimiter
	cl, err := torrent.NewClient(cfg)
	if err != nil {
		fmt.Println(err.Error())
		
		return err
	}
	defer cl.Close()
	//infoHashStr := strings.ToLower(infoHash.HexString())
	//pwd, _ := os.Getwd()
	//torrentPath := filepath.Join(pwd, infoHashStr+".torrent")
	//
	//mi, err := metainfo.LoadFromFile(torrentPath)
	//if err != nil {
	//	return err
	//}

	// 排除非法的tr地址列表;
	var newAnnounceList [][]string
	for _, announce := range infoHash.AnnounceList {
		var newTrs []string
		for _, tr := range announce {
			if !isValidTracker(tr) {
				log.Printf("种子路径: %s, 排除非法的tracker地址(1): %s", infoHash.CreatedBy, tr)
				continue
			}
			newTrs = append(newTrs, tr)
		}
		if len(newTrs) > 0 {
			newAnnounceList = append(newAnnounceList, newTrs)
		}
	}
	infoHash.AnnounceList = newAnnounceList
	// 排除非法的tr地址;
	if infoHash.Announce != "" && !isValidTracker(infoHash.Announce) {
		log.Printf("种子路径: %s, 排除非法的tracker地址(2): %s", infoHash.CreatedBy, infoHash.Announce)
		infoHash.Announce = ""
	}

	t, err := cl.AddTorrent(infoHash)
	if err != nil {
		return err
	}

	t.VerifyData()
	t.DownloadAll()
	cl.WaitAll()

	return err
}

func main() {
	metaInfo, err := metainfo.LoadFromFile("./9b8597a062f2c2f8b2dd362c28e927969287574b.torrent")
	fmt.Println(err)
	if err := AddFile(metaInfo, "/Users/jimmydang/Downloads/"); err != nil {
		fmt.Println(err)
		return
	}
	select {}
}

func isValidTracker(s string) bool {
	url_, err := url.Parse(s)
	if err != nil {
		return false
	}
	host, port, err := net.SplitHostPort(url_.Host)
	if err != nil {
		return false
	}
	if host == "" || port == "" || port == "0" {
		return false
	}
	return true
}
```
thx for reply.

**Comment by anacrolix** [comment]
Thank you for the report @jimmyduang. This issue seems related to https://github.com/anacrolix/torrent/issues/752.

**Comment by anacrolix** [comment]
I did some testing, and found that qBitTorrent will stall after a while, just like in your video. Transmission and WebTorrent do not show the same behaviour. It looks like qBitTorrent is kicking anacrolix/torrent then refusing further connections.

**Comment by anacrolix** [comment]
It looks like connections are attempted to qBitTorrent, but it closes out during or just before the handshake. I'm not sure if it's maybe due to anacrolix/torrent firing off IPv6 and IPv4 connections simultaneously, or maybe some filter on localhost connections. If I leave it running for long enough, eventually qBitTorrent accepts connections and then goes to completion. I haven't been able to view the transfer part for performance, I'm always away from my machine.

**Comment by anacrolix** [comment]
I think qBitTorrent uses libtorrent under the hood, so I'm curious if other libtorrent users are the same.

**Comment by anacrolix** [comment]
@marcovidonis was this with master, and after your latest comment in #752 ?

**Comment by marcovidonis** [comment]
@anacrolix must have been with v1.44.0, not master. I repeated the test with master now and the speed no longer drops to 0 B/s.

**Comment by anacrolix** [comment]
I'm marking this as fixed in v1.45.0. Thanks for jumping in on this @marcovidonis
